File: flutter_app/lib/data/modules.dart
import 'package:flutter/material.dart';

/// Mirrors the web app's MASTER_NAV — the full module map of EZ-Workspace.
class ModuleItem {
  const ModuleItem(this.label, this.icon);
  final String label;
  final IconData icon;
}

class ModuleSection {
  const ModuleSection(this.title, this.items);
  final String title;
  final List<ModuleItem> items;
}

const List<ModuleSection> kModules = [
  ModuleSection('Organization', [
    ModuleItem('Admin Overview', Icons.dashboard_outlined),
    ModuleItem('HR Hub', Icons.groups_outlined),
    ModuleItem('Accounts Hub', Icons.account_balance_outlined),
    ModuleItem('Projects', Icons.folder_outlined),
    ModuleItem('Employees', Icons.badge_outlined),
    ModuleItem('Shift Management', Icons.schedule_outlined),
    ModuleItem('Teams', Icons.apartment_outlined),
    ModuleItem('Org Chart', Icons.account_tree_outlined),
  ]),
  ModuleSection('Workspace', [
    ModuleItem('Workspace Hub', Icons.dashboard_customize_outlined),
    ModuleItem('Documents', Icons.description_outlined),
    ModuleItem('Spreadsheets', Icons.table_chart_outlined),
    ModuleItem('Presentations', Icons.slideshow_outlined),
    ModuleItem('Notes', Icons.sticky_note_2_outlined),
  ]),
  ModuleSection('HR & Hiring', [
    ModuleItem('Recruitment Hub', Icons.work_outline),
    ModuleItem('ATS Scanner', Icons.document_scanner_outlined),
    ModuleItem('Interviews', Icons.record_voice_over_outlined),
    ModuleItem('Onboarding', Icons.assignment_ind_outlined),
  ]),
  ModuleSection('Operations', [
    ModuleItem('Attendance', Icons.event_available_outlined),
    ModuleItem('Priority Payout', Icons.bolt_outlined),
    ModuleItem('Claims', Icons.receipt_long_outlined),
    ModuleItem('Reimbursements', Icons.payments_outlined),
    ModuleItem('Incentives', Icons.card_giftcard_outlined),
    ModuleItem('KPI / KRA', Icons.trending_up_outlined),
    ModuleItem('Payroll', Icons.currency_rupee_outlined),
    ModuleItem('Payslips', Icons.request_page_outlined),
    ModuleItem('Support Center', Icons.support_agent_outlined),
  ]),
  ModuleSection('Finance', [
    ModuleItem('Invoicing', Icons.credit_card_outlined),
    ModuleItem('Vendors', Icons.storefront_outlined),
    ModuleItem('Subscriptions', Icons.sell_outlined),
    ModuleItem('Budgets', Icons.savings_outlined),
  ]),
  ModuleSection('CRM', [
    ModuleItem('Sales Pipeline', Icons.hub_outlined),
    ModuleItem('Clients', Icons.handshake_outlined),
  ]),
  ModuleSection('Communications', [
    ModuleItem('Mail Hub', Icons.mail_outline),
    ModuleItem('Inbox', Icons.inbox_outlined),
    ModuleItem('Messages', Icons.chat_bubble_outline),
    ModuleItem('Meetings', Icons.video_camera_front_outlined),
  ]),
  ModuleSection('My Account', [
    ModuleItem('My Profile', Icons.person_outline),
    ModuleItem('My Attendance', Icons.event_note_outlined),
    ModuleItem('My Calendar', Icons.calendar_month_outlined),
    ModuleItem('My Payslips', Icons.receipt_outlined),
    ModuleItem('My Performance', Icons.speed_outlined),
    ModuleItem('Academy', Icons.school_outlined),
  ]),
  ModuleSection('System', [
    ModuleItem('Analytics', Icons.bar_chart_outlined),
    ModuleItem('Sessions', Icons.devices_outlined),
    ModuleItem('Security & Audit', Icons.gpp_good_outlined),
    ModuleItem('Permissions', Icons.admin_panel_settings_outlined),
    ModuleItem('System Config', Icons.settings_outlined),
  ]),
];

File: android_native/app/src/main/java/com/ezbillify/ezworkspace/data/Modules.kt
package com.ezbillify.ezworkspace.data

import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.outlined.Assignment
import androidx.compose.material.icons.automirrored.outlined.Chat
import androidx.compose.material.icons.automirrored.outlined.TrendingUp
import androidx.compose.material.icons.outlined.*
import androidx.compose.ui.graphics.vector.ImageVector

/** Mirrors the web app's MASTER_NAV — the full module map of EZ-Workspace. */
data class ModuleItem(val label: String, val icon: ImageVector)
data class ModuleSection(val title: String, val items: List<ModuleItem>)

val kModules: List<ModuleSection> = listOf(
    ModuleSection("Organization", listOf(
        ModuleItem("Admin Overview", Icons.Outlined.Dashboard),
        ModuleItem("HR Hub", Icons.Outlined.Groups),
        ModuleItem("Accounts Hub", Icons.Outlined.AccountBalance),
        ModuleItem("Projects", Icons.Outlined.Folder),
        ModuleItem("Employees", Icons.Outlined.Badge),
        ModuleItem("Shift Management", Icons.Outlined.Schedule),
        ModuleItem("Teams", Icons.Outlined.Apartment),
        ModuleItem("Org Chart", Icons.Outlined.AccountTree),
    )),
    ModuleSection("Workspace", listOf(
        ModuleItem("Workspace Hub", Icons.Outlined.Dashboard),
        ModuleItem("Documents", Icons.Outlined.Description),
        ModuleItem("Spreadsheets", Icons.Outlined.TableChart),
        ModuleItem("Presentations", Icons.Outlined.Slideshow),
        ModuleItem("Notes", Icons.Outlined.StickyNote2),
    )),
    ModuleSection("HR & Hiring", listOf(
        ModuleItem("Recruitment Hub", Icons.Outlined.Work),
        ModuleItem("ATS Scanner", Icons.Outlined.DocumentScanner),
        ModuleItem("Interviews", Icons.Outlined.RecordVoiceOver),
        ModuleItem("Onboarding", Icons.AutoMirrored.Outlined.Assignment),
    )),
    ModuleSection("Operations", listOf(
        ModuleItem("Attendance", Icons.Outlined.EventAvailable),
        ModuleItem("Priority Payout", Icons.Outlined.Bolt),
        ModuleItem("Claims", Icons.Outlined.ReceiptLong),
        ModuleItem("Reimbursements", Icons.Outlined.Payments),
        ModuleItem("Incentives", Icons.Outlined.CardGiftcard),
        ModuleItem("KPI / KRA", Icons.AutoMirrored.Outlined.TrendingUp),
        ModuleItem("Payroll", Icons.Outlined.CurrencyRupee),
        ModuleItem("Payslips", Icons.Outlined.RequestPage),
        ModuleItem("Support Center", Icons.Outlined.SupportAgent),
    )),
    ModuleSection("Finance", listOf(
        ModuleItem("Invoicing", Icons.Outlined.CreditCard),
        ModuleItem("Vendors", Icons.Outlined.Storefront),
        ModuleItem("Subscriptions", Icons.Outlined.Sell),
        ModuleItem("Budgets", Icons.Outlined.Savings),
    )),
    ModuleSection("CRM", listOf(
        ModuleItem("Sales Pipeline", Icons.Outlined.Hub),
        ModuleItem("Clients", Icons.Outlined.Handshake),
    )),
    ModuleSection("Communications", listOf(
        ModuleItem("Mail Hub", Icons.Outlined.Mail),
        ModuleItem("Inbox", Icons.Outlined.Inbox),
        ModuleItem("Messages", Icons.AutoMirrored.Outlined.Chat),
        ModuleItem("Meetings", Icons.Outlined.VideoCameraFront),
    )),
    ModuleSection("My Account", listOf(
        ModuleItem("My Profile", Icons.Outlined.Person),
        ModuleItem("My Attendance", Icons.Outlined.EventNote),
        ModuleItem("My Calendar", Icons.Outlined.CalendarMonth),
        ModuleItem("My Payslips", Icons.Outlined.Receipt),
        ModuleItem("My Performance", Icons.Outlined.Speed),
        ModuleItem("Academy", Icons.Outlined.School),
    )),
    ModuleSection("System", listOf(
        ModuleItem("Analytics", Icons.Outlined.BarChart),
        ModuleItem("Sessions", Icons.Outlined.Devices),
        ModuleItem("Security & Audit", Icons.Outlined.GppGood),
        ModuleItem("Permissions", Icons.Outlined.AdminPanelSettings),
        ModuleItem("System Config", Icons.Outlined.Settings),
    )),
)
